pkg/handlers/jsonl.go

```diff
@@ -147,112 +148,135 @@ func (j JSONLHandler) handleJSONLObject(f *appendable.IndexFile, r []byte, dec *
 
 			name := strings.Join(append(path, key), ".")
 
-			page, meta, err := f.FindOrCreateIndex(name, jsonTypeToFieldType(value))
-			width := meta.Width
+			for _, ft := range jsonTypeToFieldType(value) {
 
-			if err != nil {
-				return fmt.Errorf("failed to find or create index: %w", err)
-			}
-
-			mp := pointer.MemoryPointer{
-				Offset: data.Offset + uint64(fieldOffset),
-				Length: uint32(dec.InputOffset() - fieldOffset),
-			}
+				page, meta, err := f.FindOrCreateIndex(name, ft)
+				if err != nil {
+					return fmt.Errorf("failed to find or create index: %w", err)
+				}
 
-			switch value := value.(type) {
-			case string:
-				valueBytes := []byte(value)
+				width := meta.Width
 
-				if err := page.BPTree(&btree.BPTree{Data: r, DataParser: j, Width: width}).Insert(btree.ReferencedValue{
-					DataPointer: mp,
-					Value:       valueBytes,
-				}, data); err != nil {
-					return fmt.Errorf("failed to insert into b+tree: %w", err)
+				mp := pointer.MemoryPointer{
+					Offset: data.Offset + uint64(fieldOffset),
+					Length: uint32(dec.InputOffset() - fieldOffset),
 				}
-			case json.Number, float64:
-				buf := make([]byte, 8)
+
 				switch value := value.(type) {
-				case json.Number:
-					f, err := value.Float64()
-					if err != nil {
-						return fmt.Errorf("failed to parse float: %w", err)
+				case string:
+					if ft == appendable.FieldTypeString {
```

Comment:
Can this be determined by the meta page?

Reply:
Yeah, I was unsure if this was the best approach, so I decided to not touch it.

`jsonTypeToFieldType` returns a list of valid `FieldTypes` given the `value.(type)` If `value` is a `string`, it causes a variance in behavior. So when we loop by the different `fieldType`, I figured I'd add some guard to the logic below.

This leads to a lot of tests breaking which should occur since every string var should cause two indexes to form, but should I case by the fieldType instead of `value.(type)`?

Reply:
Yeah it makes more sense to case on field type. Basically any type introspection at the handler level is left over from back when we had trees that stored multiple types. It's no longer necessary and in retrospect that was a bad idea so it's good we got rid of it :)

Reply:
Sounds good. Note, we'll still have to do some `value` type introspection. Something like:

```go
switch ft {
     case FieldTypeTrigram:
          switch (value).type {
                  string: // do stuff
                  default: return fmt.Errorf("wrong type")
          }
}
```

Or is this roundabout, and we can directly cast to a string since we made the assertions above?

Reply:
We can directly cast to a string, something like:

```go
switch ft {
  case FieldTypeTrigram:
    value, ok := value.(string)
    if !ok {
      return fmt.Errorf("expected a string didn't get a string")
    }
}
```